### ISSUE TITLE
Crypto public key has wrong panic message

### ISSUE THREAD

**Issue by andrewnguyen22** [pokt-network/posmint#116]
panic("unsupported private key type")

**Comment by andrewnguyen22** [comment]
Also should support native types:

```
func PubKeyToPublicKey(key crypto.PubKey) PublicKey {
	switch key.(type) {
	case secp256k1.PubKeySecp256k1:
		return Secp256k1PublicKey{}.PubKeyToPublicKey(key)
	case ed25519.PubKeyEd25519:
		return Ed25519PublicKey{}.PubKeyToPublicKey(key)
	case Ed25519PublicKey:
		return Ed25519PublicKey{}.PubKeyToPublicKey(key)
	case Secp256k1PublicKey:
		return Secp256k1PublicKey{}.PubKeyToPublicKey(key)
	}
	panic("unsupported private key type")
}
```

**Comment by andrewnguyen22** [comment]
Scratch that ^^

**Comment by andrewnguyen22** [comment]
closed with #118
